integr8ly/integreatly-operator
# [wip] use cro to generate smtp creds for 3scale

currently, we have to set the smtp credentials in the 3scale
cluster manually after an rhmi installation has been completed.

this change leverages the cloud resource operator to instead
create the smtp credentials required in a cloud provider.

- add a new useExternalResources key to the installation cr spec
- if useExternalResources is enabled
  - create the smtpcredentialset cr
  - wait for it to be complete
  - use the smtp credentials in the smtp configmap for 3scale

verification:
- have an openshift 4 cluster
- have the cloud resource operator setup in the same namespace
- create an installation cr with useExternalResources set to true
- ensure the smtp configmap in the 3scale namespace is populated
- in 3scale, try to invite a new user, an email should be sent
